diff --git a/bookmarks/queries.py b/bookmarks/queries.py
index 17a7a35..0fa35de 100644
--- a/bookmarks/queries.py
+++ b/bookmarks/queries.py
@@ -1,4 +1,5 @@
 from django.contrib.auth.models import User
+from django.contrib.postgres.aggregates import StringAgg
 from django.db.models import Q, Count, Aggregate, CharField, Value, BooleanField, QuerySet
 
 from bookmarks.models import Bookmark, Tag
@@ -31,7 +32,7 @@ def _base_bookmarks_query(user: User, query_string: str) -> QuerySet:
     # Add aggregated tag info to bookmark instances
     query_set = Bookmark.objects \
         .annotate(tag_count=Count('tags'),
-                  tag_string=Concat('tags__name'),
+                  tag_string=StringAgg('tags__name', delimiter=','),
                   tag_projection=Value(True, BooleanField()))
 
     # Filter for user
